src/en/comicfans/src/eu/kanade/tachiyomi/extension/en/comicfans/ComicFans.kt

```diff
@@ -0,0 +1,202 @@
+                with(element.selectFirst(".book-name > a")!!) {
+                    title = text()
+                    setUrlWithoutDomain(attr("abs:href"))
+                }
+            }
+        }
+
+        return MangasPage(mangaList, false)
+    }
+    // =============================== Search ===============================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/books".toHttpUrl().newBuilder().apply {
+            addQueryParameter("pageNumber", page.toString())
+            addQueryParameter("pageSize", "20")
+            fragment(page.toString())
+
+            if (query.isNotBlank()) {
+                addPathSegment("search")
+                addQueryParameter("keyWord", query)
+            } else {
+                filters.getUriPart<GenreFilter>()?.let {
+                    addQueryParameter("genre", it)
+                }
+                filters.getUriPart<LastUpdateFilter>()?.let {
+                    addQueryParameter("withinDay", it)
+                }
+                filters.getUriPart<StatusFilter>()?.let {
+                    addQueryParameter("status", it)
+                }
+            }
+        }.build()
+
+        val searchHeaders = apiHeadersBuilder()
+            .set("Accept", "*/*")
+            .build()
+
+        return GET(url, searchHeaders)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage =
+        popularMangaParse(response)
+
+    // =============================== Filters ==============================
+
+    override fun getFilterList(): FilterList = FilterList(
+        Filter.Header("Text search ignores filters"),
+        Filter.Separator(),
+        GenreFilter(),
+        LastUpdateFilter(),
+        StatusFilter(),
+    )
+
+    // =========================== Manga Details ============================
+
+    override fun mangaDetailsParse(response: Response): SManga {
+        return response.parseData<MangaDto>().toSManga(cdnUrl)
+    }
+
+    // ============================== Chapters ==============================
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val chapterHeaders = apiHeadersBuilder()
+            .set("Accept", "*/*")
+            .build()
+
+        val bookId = manga.url.substringAfter("/comic/")
+            .substringBefore("-")
+
+        return GET("$apiUrl/chapters/page?sortDirection=ASC&bookId=$bookId&pageNumber=1&pageSize=9999", chapterHeaders)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        return response.parseAs<ChapterListDto>().data.list.mapIndexed { index, chapterDto ->
+            chapterDto.toSChapter(index + 1)
+        }.reversed()
+    }
+
+    // =============================== Pages ================================
+
+    override fun pageListParse(response: Response): List<Page> {
+        return response.parseData<Data2Dto>().comicImageList.map {
```

Comment:
I think they have api endpoint for pages too
`https://api.comicfans.io/comic-backend/api/v1/content/chapters/414`
`$apiUrl/chapters/$chapter_id`

Reply:
Yup, that seems to work. How did you find this? Just guessing? I never saw it when looking at the requests made by the site